test/src/main/java/org/apache/accumulo/test/fate/accumulo/AccumuloStoreReadWriteIT.java

```diff
@@ -119,6 +126,82 @@ public void testReadWrite() throws Exception {
     }
   }
 
+  @Test
+  public void testDeferredOverflow() throws Exception {
+    final String table = getUniqueNames(1)[0];
+    try (ClientContext client =
+        (ClientContext) Accumulo.newClient().from(getClientProps()).build()) {
+      client.tableOperations().create(table);
+
+      AccumuloStore<TestEnv> store = new AccumuloStore<>(client, table, 10);
+      // Verify no transactions
+      assertEquals(0, store.list().count());
+      assertFalse(store.isDeferredOverflow());
+
+      // Store 10 transactions that are all deferred
+      final Set<Long> transactions = new HashSet<>();
+      for (int i = 0; i < 10; i++) {
+        long tid = store.create();
+        transactions.add(tid);
+        FateTxStore<TestEnv> txStore = store.reserve(tid);
+        txStore.setStatus(TStatus.SUBMITTED);
+        assertTrue(txStore.timeCreated() > 0);
+        txStore.unreserve(10, TimeUnit.SECONDS);
+      }
+
+      // Verify we have 10 transactions and all are deferred
+      assertEquals(10, store.list().count());
+      assertEquals(10, store.getDeferredCount());
+
+      // Should still be false as we are at thet max but not over yet
+      assertFalse(store.isDeferredOverflow());
+
+      AtomicBoolean keepRunning = new AtomicBoolean(true);
+      // Run and verify all 10 transactions still exist and were not
+      // run because of the deferral time of all the transactions
+      try {
+        ForkJoinPool.commonPool().execute(() -> store.runnable(keepRunning, transactions::remove));
```

Comment:
The test does not seem to know if this task is still running in the background or not.  Seems like it could run concurrently with later test code.  

Also I am uncertain about using this JVM wide thread pool, if other accumulo code or a third party library throws blocking task on it then the task in this test may not run.